neumai/neumai/EmbedConnectors/HuggingfaceEmbedder.py

```diff
@@ -0,0 +1,51 @@
+from typing import List, Tuple
+from neumai.EmbedConnectors.EmbedConnector import EmbedConnector
+from neumai.Shared.NeumDocument import NeumDocument
+from huggingface_hub import InferenceClient
+from pydantic import Field
+
+class HuggingFaceEmbed(EmbedConnector):
+    """
+    Hugging Face Embed Connector
+    """
+    model:str = Field(..., description="HuggingFace model ID or a URL to a deployed Inference Endpoint")
+    token: str = Field(..., description="HuggingFace token")
+
+    @property
+    def embed_name(self) -> str:
+        return 'HuggingFaceEmbed'
+    
+    @property
+    def required_properties(self) -> List[str]:
+        return ["token"]
+
+    @property
+    def optional_properties(self) -> List[str]:
+        return []
+    
+    def validation(self) -> bool:
+        """config_validation connector setup"""
```

Comment:
Any validation steps that can help make sure config is correct?

Reply:
Would it be enough to initiate the inferenceClient with parameters?

Reply:
Yeah that would be great. We just want to ensure we can validate connections before we start processing data.